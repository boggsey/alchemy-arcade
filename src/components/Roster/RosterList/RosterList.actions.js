import { error } from 'react-notification-system-redux';
import {
  ROSTER_LIST_REQUEST,
  ROSTER_LIST_SUCCESS,
  ROSTER_LIST_FAILURE,
} from './RosterList.constants';

function requestRosterList() {
  return {
    type: ROSTER_LIST_REQUEST,
  };
}

function receiveRosterList(players) {
  return {
    type: ROSTER_LIST_SUCCESS,
    payload: players,
  };
}

function errorRosterList() {
  return {
    type: ROSTER_LIST_FAILURE,
  };
}

async function rosterListRequest(token) {
  const response = await fetch('https://players-api.developer.alchemy.codes/api/players/', {
    headers: {
      Authorization: `Bearer ${token}`,
    },
    method: 'GET',
  }).catch((err) => {
    throw new Error(err.message);
  });
  return response.json();
}

const getRosterList = (token) => { 
  return async (dispatch) => {
    dispatch(requestRosterList());
    try {
      const data = await rosterListRequest(token);
      if (data.success) {
        dispatch(receiveRosterList(data.players));
      } else {
        dispatch(errorRosterList());
        dispatch(error({ position: 'tr', message: data.error.message, autoDismiss: 10 }));
      }
    } catch (err) {
      dispatch(errorRosterList());
      dispatch(error(err));
    }
  };
};

export default getRosterList;
