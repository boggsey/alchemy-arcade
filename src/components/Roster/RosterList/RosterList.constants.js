export const ROSTER_LIST_REQUEST = 'ROSTER_LIST_REQUEST';
export const ROSTER_LIST_SUCCESS = 'ROSTER_LIST_SUCCESS';
export const ROSTER_LIST_FAILURE = 'ROSTER_LIST_FAILURE';
