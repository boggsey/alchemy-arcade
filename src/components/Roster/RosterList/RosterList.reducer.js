import {
  ROSTER_LIST_REQUEST,
  ROSTER_LIST_SUCCESS,
  ROSTER_LIST_FAILURE,
} from './RosterList.constants';

const initialState = {
  players: [],
};

function RosterListReducer(state = initialState, action) {
  switch (action.type) {
    case ROSTER_LIST_REQUEST:
      return {
        ...state,
      };
    case ROSTER_LIST_SUCCESS:
      return {
        ...state,
        players: [...action.payload],
      };
    case ROSTER_LIST_FAILURE:
      return {
        ...state,
      };
    default:
      return state;
  }
}

export default RosterListReducer;
