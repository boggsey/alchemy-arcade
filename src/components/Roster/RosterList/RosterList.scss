@import '../../../styles/variables.scss';

.roster_list-wrapper {
  width: 60%;
  margin: 0 auto;
  display: flex;
  flex-direction: column;

  @media only screen and (max-width: 420px) {
    width: 95%;
  }

  @media only screen and (max-width: 1024px) {
    width: 80%;
  }
}

.add-new {
  color: $alchemy-green;
  border: 2px solid $alchemy-green;
  font-family: 'Press Start 2P';
  font-size: 10px;
  padding: 10px;
  text-decoration: none;
  margin-left: auto;
}

.ReactTable {
  background: transparent;
  height: 300px;
  width: 100%;
  border: 0;
  font-family: 'Press Start 2P';
  font-size: 10px;

  @media only screen and (max-width: 420px) {
    width: 100%;
    min-width: 100%;
    height: auto;
  }

  @media only screen and (max-width: 1024px) {
    width: 100%;
  } 

  .rt-table {
    overflow: visible;
  }

  .rt-tr {
    display: flex;
    
    @media only screen and (max-width: 420px) {
      flex-flow: row wrap;
      padding: 10px 0;
      align-items: center;
    }
  }

  .-headerGroups {
    display: none;
  }

  .rt-thead {
    @media only screen and (max-width: 420px) {
      display: none;
    }

    &.-header {
      box-shadow: none;
    }

    .rt-th {
      color: #000;
      background: $alchemy-green;
      border: 0;
      box-shadow: 0;
      padding: 20px 0;
      margin: 10px 0 10px 0;

      &.-sort-asc {
        box-shadow: none;
      }

      &.-sort-desc {
        box-shadow: none;
      }
    }
  }

  .rt-tbody {
    @media only screen and (max-width: 420px) {
      min-width: 100% !important;
      width: 100%;
      flex: none;
    }

    .rt-tr-group {
      border: 0;

      @media only screen and (max-width: 420px) {
        min-width: 100%;
        width: 100%;
      }
    }

    .tr-tr {
      display: flex;
      flex-direction: column !important;

      @media only screen and (max-width: 420px) {
        min-width: 100%;
        width: 100%;
      }
    }

    .rt-td {
      border-right: 0;
      color: $alchemy-green;
      text-transform: capitalize;
      text-align: center;
    }

    .-rating {
      @media only screen and (max-width: 420px) {
        width: 100% !important;
        color: #000;
        background: $alchemy-green;
      }
    }
  }

  .pagination-bottom {
    height: 50px;
  
    .-pagination {
      height: 50px;
      box-shadow: none;
      border: 0;

      .-btn {
        color: $alchemy-green;
      }
  
      .-center {
        display: none;
      }
    }
  }

  .delete {
    cursor: pointer;
    text-decoration: underline;
  }
}
