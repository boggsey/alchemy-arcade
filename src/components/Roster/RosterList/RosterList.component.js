import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactTable from 'react-table';
import 'react-table/react-table.css';
import getRosterList from './RosterList.actions';
import RosterDelete from '../RosterDelete/RosterDelete.component';
import './RosterList.scss';

export class RosterList extends Component {

  componentDidMount() {
    this.getTableData();
  }

  async getTableData() {
    const token = window.localStorage.getItem('token');
    try {
      await this.props.getRosterList(token);
    } catch (error) {
      console.log(error);
    }
  }

  render() {
    return (
      <ReactTable
        data={this.props.players}
        noDataText="Take me to your leader"
        defaultSorted={[
          {
            id: 'rating',
            desc: true,
          },
        ]}
        resizable={false}
        sortable={false}
        showPageSizeOptions={false}
        columns={[
          {
            columns: [
              {
                Header: 'First',
                accessor: 'first_name',
              },
              {
                Header: 'Last',
                id: 'last_name',
                accessor: 'last_name',
              },
              {
                Header: 'Rating',
                id: 'rating',
                accessor: 'rating',
              },
              {
                Header: 'Hand',
                id: 'handedness',
                accessor: 'handedness',
              },
              {
                Header: 'Actions',
                id: 'actions',
                accessor: 'id',
                Cell: row => (
                  <RosterDelete playerId={row.value} />
                ),
              },
            ],
          },
        ]}
        defaultPageSize={5}
      />
    );
  }
}

RosterList.propTypes = {
  players: PropTypes.arrayOf(PropTypes.object).isRequired,
  getRosterList: PropTypes.func.isRequired,
};

const mapStateToProps = state => ({
  players: state.RosterListReducer.players,
});

const mapDispatchToProps = {
  getRosterList,
};

export default connect(mapStateToProps, mapDispatchToProps)(RosterList);
