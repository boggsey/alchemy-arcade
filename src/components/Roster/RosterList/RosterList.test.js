import React from 'react';
import { shallow } from 'enzyme';
import fakeStore from '../../../fakeStore';
import { RosterList } from './RosterList.component';

const state = { RosterListReducer: {} };

describe('RosterList', () => {
  it('Hasn\'t changed since the last snapshot', () => {
    const wrapper = shallow(
      <RosterList store={fakeStore} state={state} players={[]} getRosterList={() => {}} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
