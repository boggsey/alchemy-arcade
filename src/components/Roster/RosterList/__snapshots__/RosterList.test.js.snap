// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RosterList Hasn't changed since the last snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RosterList
    getRosterList={[Function]}
    players={Array []}
    state={
      Object {
        "RosterListReducer": Object {},
      }
    }
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "subscribe": [Function],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "AggregatedComponent": [Function],
      "ExpanderComponent": [Function],
      "FilterComponent": [Function],
      "LoadingComponent": [Function],
      "NextComponent": undefined,
      "NoDataComponent": [Function],
      "PadRowComponent": [Function],
      "PaginationComponent": [Function],
      "PivotComponent": undefined,
      "PivotValueComponent": [Function],
      "PreviousComponent": undefined,
      "ResizerComponent": [Function],
      "TableComponent": [Function],
      "TbodyComponent": [Function],
      "TdComponent": [Function],
      "TfootComponent": [Function],
      "ThComponent": [Function],
      "TheadComponent": [Function],
      "TrComponent": [Function],
      "TrGroupComponent": [Function],
      "aggregatedKey": "_aggregated",
      "className": "",
      "collapseOnDataChange": true,
      "collapseOnPageChange": true,
      "collapseOnSortingChange": true,
      "column": Object {
        "Aggregated": undefined,
        "Cell": undefined,
        "Expander": undefined,
        "Filter": undefined,
        "Footer": undefined,
        "Header": undefined,
        "Pivot": undefined,
        "PivotValue": undefined,
        "aggregate": undefined,
        "className": "",
        "filterAll": false,
        "filterMethod": undefined,
        "filterable": undefined,
        "footerClassName": "",
        "footerStyle": Object {},
        "getFooterProps": [Function],
        "getHeaderProps": [Function],
        "getProps": [Function],
        "headerClassName": "",
        "headerStyle": Object {},
        "minWidth": 100,
        "resizable": undefined,
        "show": true,
        "sortMethod": undefined,
        "sortable": undefined,
        "style": Object {},
      },
      "columns": Array [
        Object {
          "columns": Array [
            Object {
              "Header": "First",
              "accessor": "first_name",
            },
            Object {
              "Header": "Last",
              "accessor": "last_name",
              "id": "last_name",
            },
            Object {
              "Header": "Rating",
              "accessor": "rating",
              "id": "rating",
            },
            Object {
              "Header": "Hand",
              "accessor": "handedness",
              "id": "handedness",
            },
            Object {
              "Cell": [Function],
              "Header": "Actions",
              "accessor": "id",
              "id": "actions",
            },
          ],
        },
      ],
      "data": Array [],
      "defaultExpanded": Object {},
      "defaultFilterMethod": [Function],
      "defaultFiltered": Array [],
      "defaultPageSize": 5,
      "defaultResized": Array [],
      "defaultSortDesc": false,
      "defaultSortMethod": [Function],
      "defaultSorted": Array [
        Object {
          "desc": true,
          "id": "rating",
        },
      ],
      "expanderDefaults": Object {
        "filterable": false,
        "resizable": false,
        "sortable": false,
        "width": 35,
      },
      "filterable": false,
      "freezeWhenExpanded": false,
      "getLoadingProps": [Function],
      "getNoDataProps": [Function],
      "getPaginationProps": [Function],
      "getProps": [Function],
      "getResizerProps": [Function],
      "getTableProps": [Function],
      "getTbodyProps": [Function],
      "getTdProps": [Function],
      "getTfootProps": [Function],
      "getTfootTdProps": [Function],
      "getTfootTrProps": [Function],
      "getTheadFilterProps": [Function],
      "getTheadFilterThProps": [Function],
      "getTheadFilterTrProps": [Function],
      "getTheadGroupProps": [Function],
      "getTheadGroupThProps": [Function],
      "getTheadGroupTrProps": [Function],
      "getTheadProps": [Function],
      "getTheadThProps": [Function],
      "getTheadTrProps": [Function],
      "getTrGroupProps": [Function],
      "getTrProps": [Function],
      "groupedByPivotKey": "_groupedByPivot",
      "indexKey": "_index",
      "loading": false,
      "loadingText": "Loading...",
      "multiSort": true,
      "nestingLevelKey": "_nestingLevel",
      "nextText": "Next",
      "noDataText": "Take me to your leader",
      "ofText": "of",
      "onExpandedChange": undefined,
      "onFetchData": [Function],
      "onFilteredChange": undefined,
      "onPageChange": undefined,
      "onPageSizeChange": undefined,
      "onResizedChange": undefined,
      "onSortedChange": undefined,
      "originalKey": "_original",
      "pageSizeOptions": Array [
        5,
        10,
        20,
        25,
        50,
        100,
      ],
      "pageText": "Page",
      "pivotBy": undefined,
      "pivotDefaults": Object {},
      "pivotIDKey": "_pivotID",
      "pivotValKey": "_pivotVal",
      "previousText": "Previous",
      "resizable": false,
      "resolveData": [Function],
      "rowsText": "rows",
      "showPageJump": true,
      "showPageSizeOptions": false,
      "showPagination": true,
      "showPaginationBottom": true,
      "showPaginationTop": false,
      "sortable": false,
      "style": Object {},
      "subRowsKey": "_subRows",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "AggregatedComponent": [Function],
        "ExpanderComponent": [Function],
        "FilterComponent": [Function],
        "LoadingComponent": [Function],
        "NextComponent": undefined,
        "NoDataComponent": [Function],
        "PadRowComponent": [Function],
        "PaginationComponent": [Function],
        "PivotComponent": undefined,
        "PivotValueComponent": [Function],
        "PreviousComponent": undefined,
        "ResizerComponent": [Function],
        "TableComponent": [Function],
        "TbodyComponent": [Function],
        "TdComponent": [Function],
        "TfootComponent": [Function],
        "ThComponent": [Function],
        "TheadComponent": [Function],
        "TrComponent": [Function],
        "TrGroupComponent": [Function],
        "aggregatedKey": "_aggregated",
        "className": "",
        "collapseOnDataChange": true,
        "collapseOnPageChange": true,
        "collapseOnSortingChange": true,
        "column": Object {
          "Aggregated": undefined,
          "Cell": undefined,
          "Expander": undefined,
          "Filter": undefined,
          "Footer": undefined,
          "Header": undefined,
          "Pivot": undefined,
          "PivotValue": undefined,
          "aggregate": undefined,
          "className": "",
          "filterAll": false,
          "filterMethod": undefined,
          "filterable": undefined,
          "footerClassName": "",
          "footerStyle": Object {},
          "getFooterProps": [Function],
          "getHeaderProps": [Function],
          "getProps": [Function],
          "headerClassName": "",
          "headerStyle": Object {},
          "minWidth": 100,
          "resizable": undefined,
          "show": true,
          "sortMethod": undefined,
          "sortable": undefined,
          "style": Object {},
        },
        "columns": Array [
          Object {
            "columns": Array [
              Object {
                "Header": "First",
                "accessor": "first_name",
              },
              Object {
                "Header": "Last",
                "accessor": "last_name",
                "id": "last_name",
              },
              Object {
                "Header": "Rating",
                "accessor": "rating",
                "id": "rating",
              },
              Object {
                "Header": "Hand",
                "accessor": "handedness",
                "id": "handedness",
              },
              Object {
                "Cell": [Function],
                "Header": "Actions",
                "accessor": "id",
                "id": "actions",
              },
            ],
          },
        ],
        "data": Array [],
        "defaultExpanded": Object {},
        "defaultFilterMethod": [Function],
        "defaultFiltered": Array [],
        "defaultPageSize": 5,
        "defaultResized": Array [],
        "defaultSortDesc": false,
        "defaultSortMethod": [Function],
        "defaultSorted": Array [
          Object {
            "desc": true,
            "id": "rating",
          },
        ],
        "expanderDefaults": Object {
          "filterable": false,
          "resizable": false,
          "sortable": false,
          "width": 35,
        },
        "filterable": false,
        "freezeWhenExpanded": false,
        "getLoadingProps": [Function],
        "getNoDataProps": [Function],
        "getPaginationProps": [Function],
        "getProps": [Function],
        "getResizerProps": [Function],
        "getTableProps": [Function],
        "getTbodyProps": [Function],
        "getTdProps": [Function],
        "getTfootProps": [Function],
        "getTfootTdProps": [Function],
        "getTfootTrProps": [Function],
        "getTheadFilterProps": [Function],
        "getTheadFilterThProps": [Function],
        "getTheadFilterTrProps": [Function],
        "getTheadGroupProps": [Function],
        "getTheadGroupThProps": [Function],
        "getTheadGroupTrProps": [Function],
        "getTheadProps": [Function],
        "getTheadThProps": [Function],
        "getTheadTrProps": [Function],
        "getTrGroupProps": [Function],
        "getTrProps": [Function],
        "groupedByPivotKey": "_groupedByPivot",
        "indexKey": "_index",
        "loading": false,
        "loadingText": "Loading...",
        "multiSort": true,
        "nestingLevelKey": "_nestingLevel",
        "nextText": "Next",
        "noDataText": "Take me to your leader",
        "ofText": "of",
        "onExpandedChange": undefined,
        "onFetchData": [Function],
        "onFilteredChange": undefined,
        "onPageChange": undefined,
        "onPageSizeChange": undefined,
        "onResizedChange": undefined,
        "onSortedChange": undefined,
        "originalKey": "_original",
        "pageSizeOptions": Array [
          5,
          10,
          20,
          25,
          50,
          100,
        ],
        "pageText": "Page",
        "pivotBy": undefined,
        "pivotDefaults": Object {},
        "pivotIDKey": "_pivotID",
        "pivotValKey": "_pivotVal",
        "previousText": "Previous",
        "resizable": false,
        "resolveData": [Function],
        "rowsText": "rows",
        "showPageJump": true,
        "showPageSizeOptions": false,
        "showPagination": true,
        "showPaginationBottom": true,
        "showPaginationTop": false,
        "sortable": false,
        "style": Object {},
        "subRowsKey": "_subRows",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
